### ISSUE TITLE
Pytmx throws KeyError: 'class' with new maps.

### ISSUE THREAD

**Issue by SoupySoups** [bitcraft/pytmx#160]
Due to the new custom classes in Tiled 1.8+ pytmx throws `KeyError: 'class'` whilst parsing a map with [custom classes](https://doc.mapeditor.org/en/stable/manual/custom-properties/#custom-property-types) / [enumerators](https://doc.mapeditor.org/en/stable/manual/custom-properties/#custom-enums).

Traceback:
```
  File "c:\***\src\managers\data_manager.py", line 31, in __init__
    self.raw_tmxdata = pytmx.load_pygame(self.filename)
  File "C:\***\venv\lib\site-packages\pytmx\util_pygame.py", line 183, in load_pygame
    return pytmx.TiledMap(filename, *args, **kwargs)
  File "C:\***\venv\lib\site-packages\pytmx\pytmx.py", line 497, in __init__
    self.parse_xml(ElementTree.parse(self.filename).getroot())
  File "C:\***\venv\lib\site-packages\pytmx\pytmx.py", line 539, in parse_xml
    objectgroup = TiledObjectGroup(self, subnode)
  File "C:\***\venv\lib\site-packages\pytmx\pytmx.py", line 1357, in __init__
    self.parse_xml(node)
  File "C:\***\venv\lib\site-packages\pytmx\pytmx.py", line 1368, in parse_xml
    self.extend(TiledObject(self.parent, child) for child in node.findall("object"))
  File "C:\***\venv\lib\site-packages\pytmx\pytmx.py", line 1368, in <genexpr>
    self.extend(TiledObject(self.parent, child) for child in node.findall("object"))
  File "C:\***\venv\lib\site-packages\pytmx\pytmx.py", line 1397, in __init__
    self.parse_xml(node)
  File "C:\***\venv\lib\site-packages\pytmx\pytmx.py", line 1427, in parse_xml
    self._set_properties(node)
  File "C:\***\venv\lib\site-packages\pytmx\pytmx.py", line 396, in _set_properties
    properties = parse_properties(node)
  File "C:\***\venv\lib\site-packages\pytmx\pytmx.py", line 320, in parse_properties
    cls = prop_type[subnode.get("type")]
KeyError: 'class'
```
Tested Tiled version: `1.8.4`
Tested PyTMX version: `3.31` (latest)

This issue seems to be a outdated issue similar to #108.

**Comment by bitcraft** [comment]
ty for linking the documentation.  are you able to provide the map file for reference?  it would help to get this implemented correctly.

**Comment by SoupySoups** [comment]
Github wont let me upload the tmx or json files within this issue but here is a zip of them [levels.zip](https://github.com/bitcraft/pytmx/files/8457616/levels.zip). It seems as though the map file does not ever reference the properties json file witch is seems like it is going to be very inconvenient to load. might open a Tiled issue on this.

**Comment by SoupySoups** [comment]
I talked to the tiled developers and they are agreeing and saying they are gonna try to embed the custom properties within the map file itself rather than keeping the `propertytypes.json`

**Comment by bitcraft** [comment]
ty for the context and the zip file.  thats super helpful.  so currently, propertytypes.json is required, but at some time won't be?

**Comment by SoupySoups** [comment]
Yes that is correct

**Comment by bitcraft** [comment]
perfect, thanks

**Comment by SoupySoups** [comment]
Created a pull request with an updated `pytmx.py` for 1.8.x+ #161

**Comment by bitcraft** [comment]
closed by #161
